patches/tModLoader/Terraria/ModLoader/LocalizationLoader.cs

```diff
@@ -621,6 +621,13 @@ private static LocalizationFile FindHJSONFileForKey(List<LocalizationFile> files
 				levelFound = level;
 				best = file;
 			}
+			if (!string.IsNullOrWhiteSpace(file.prefix)) {
+				level = file.prefix.Split(".").Length;
+				if (level > levelFound) {
+					levelFound = level;
+					best = file;
+				}
+			}
```

Comment:
would be cleaner to factor this into `LongestMatchingPrefix`, it could just take the `LocalizationFile` as an arg instead of `entries` and start with for loop at `file.Prefix.Split(".").Length`